docs/design/scheduler.md

```diff
@@ -0,0 +1,46 @@
+
+# The Kubernetes Scheduler
+
+The Kubernetes scheduler runs as a process alongside the other master
+components such as the API server. Its interface to the API server is to watch
+for Pods with an empty PodSpec.NodeName, and for each them to post a Binding
+indicating where the Pod should be scheduled.
+
+## The scheduling process
+
+The scheduler tries to find a node for each Pod, one at a time, as it notices
+these Pods via watch. There are three steps. First it applies a set of "predicates" that filter out
+inappropriate nodes. For example, if the PodSpec specifies resource limits, then the scheduler
+will filter out nodes that don't have at least that much resources available (computed
+as the capacity of the node minus the sum of the resource limits of the Pods that
+are already running on the node). Second, it applies a set of "priority functions"
+that rank the nodes that weren't filtered out by the predicate check. For example,
+if the PodSpec specified resource limits, then it favors the least-loaded
+machine (measured in terms of the sum of the limits of the Pods already on each node).
+Finally, the node with the highest priority is chosen
```

Comment:
May we should say something about weight?

Each priority function is assigned with a weight (a positive number) representing how much this priority function would affected the final score when ranking each node.